### ISSUE TITLE
Unclear size of FWFT Feature Types (Chapter 18)

### ISSUE THREAD

**Issue by sfiedler0** [riscv-non-isa/riscv-sbi-doc#174]
In [the text](https://github.com/riscv-non-isa/riscv-sbi-doc/blob/master/src/ext-firmware-features.adoc?plain=1#L3-L5) the FWFT feature types are described as "32-bit identifiers", but in [function calls](https://github.com/riscv-non-isa/riscv-sbi-doc/blob/master/src/ext-firmware-features.adoc?plain=1#L110-L112) they are described as `unsigned long`, which will result in 64-bit identifiers.

**Comment by atishp04** [comment]
Good catch.
@clementleger @avpatel : We don't need 64 bit feature values. We should change the function arguments to uint32_t ? It would require the fixes in the existing implementation though.

**Comment by clementleger** [comment]
Yes clearly that should be defined as `uint32_t`, thanks for catching this. I'll submit fixes for the doc/implementation if you want.

**Comment by jrtc27** [comment]
The documentation says:

> Every SBI function should prefer unsigned long as the data type. It keeps the specification simple and easily adaptable for all RISC-V ISA types. In case the data is defined as 32bit wide, higher privilege software must ensure that it only uses 32 bit data.

This doesn't read very clearly to me. Does that mean the function should be declared as taking an unsigned long but only read 32 bits from it? Or does it mean the function should be declared as taking a 32-bit integer? Also it seems to suggest that the high bits of the register on RV64 can be junk, which is notably *not* what the psABI says, so you can't just forward the register in assembly to a C function taking a uint32_t.

**Comment by jrtc27** [comment]
Though it seems there are a number of interfaces already taking uint32_t and hitting that exact case of a lack of adequate documentation...

**Comment by sfiedler0** [comment]
I think we can close this issue now.
